vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ModuleManager.java

```diff
@@ -89,4 +99,32 @@ public ISequence checkIfSequenceExists(String moduleId, String sequenceId) {
     public Page<IModule> findByNameOrDescription(Pageable requestedPage,String searchText) {
         return moduleRepo.findDistinctByNameContainingOrDescriptionContaining(requestedPage,searchText,searchText);
     }
+    
+    @Override
+    public void deleteModule(String moduleId) {
+        List<ModuleLink> moduleLinksToPersist = new ArrayList<ModuleLink>();
+        if(moduleId != null) {
+            List<IModuleLink> moduleLinks = moduleLinkRepo.findModuleLinkByModuleId(moduleId);
+            for(IModuleLink moduleLink: moduleLinks) {
+                moduleLink.setModule(null);
+                moduleLinksToPersist.add((ModuleLink)moduleLink);
```

Comment:
is there a reason you're not using the `moduleLinks` list and persist it but create a new list?

Reply:
changed